libraries/AP_SmartAudio/AP_SmartAudio.h

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
+#include <AP_Param/AP_Param.h>
+#include <stdio.h>
+
+#ifndef SMARTAUDIO_ENABLED
+#define SMARTAUDIO_ENABLED !HAL_MINIMIZE_FEATURES && !APM_BUILD_TYPE(APM_BUILD_Replay)
+#endif
+
+#if SMARTAUDIO_ENABLED
+#define SMARTAUDIO_BUFFER_CAPACITY 5
+
+#include "smartaudio_protocol.h"
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_HAL/utility/RingBuffer.h>
+#include <AP_RCTelemetry/AP_VideoTX.h>
+
+
+class AP_SmartAudio
+{
+
+public:
+
+    // request packet to be processed
+    struct packet{
+        smartaudioFrame_t frame;
+        uint8_t frame_size=0;
```

Comment:
Please don't initialize structures or member variables with null defaults - it messes up the bss, rely on the C++ rules for initialization.

Reply:
That's all the rules for initialization that i must consider ? http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-always